web/federate.go

```diff
@@ -0,0 +1,98 @@
+package web
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"bitbucket.org/ww/goautoneg"
+	"github.com/golang/protobuf/proto"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/text"
+
+	"github.com/prometheus/prometheus/promql"
+
+	clientmodel "github.com/prometheus/client_golang/model"
+	dto "github.com/prometheus/client_model/go"
+)
+
+type Federation struct {
+	QueryEngine *promql.Engine
+}
+
+func (fed *Federation) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	req.ParseForm()
+	qs := fmt.Sprintf("{%s}", strings.Join(req.Form["match[]"], ","))
```

Comment:
This could be injected, and lead to use cases we don't want to support.

Could we pass this more directly to the parser?
Being able to specify multiple matches that are ORed together would be handy.